### PR TITLE
WT-10905 Fix update reconstruction for read-uncommitted isolation

### PR DESCRIPTION
**Pull request by korteland**
See the ticket for the Python test I used to verify this. That's not appropriate to add to the repo since it relies on adding a sleep, so the failure config is included here.

There's a comment describing the bug in more depth, so I won't repeat it here.

### CODE REVIEW

File: src/support/modify.c [hunk 1]
```diff
@@ -470,8 +470,7 @@ __wt_modify_reconstruct_from_upd_list(
     __wt_update_vector_init(session, &modifies);
     /* Find a complete update. */
     for (upd = modify; upd != NULL; upd = upd->next) {
-        if (upd->txnid == WT_TXN_ABORTED)
-            continue;
+        WT_ASSERT(session, upd->type != WT_UPDATE_TOMBSTONE);
```
Comment (quchenhao) [inline src/support/modify.c:473]: Shall we also assert the update is not aborted?

File: src/support/modify.c [hunk 2]
```diff
@@ -470,8 +470,7 @@ __wt_modify_reconstruct_from_upd_list(
...
 
         if (WT_UPDATE_DATA_VALUE(upd))
```
Comment (quchenhao) [inline src/support/modify.c:475]: Because we cannot see tombstone, this check can be simplified to only check a full value.

Comment (quchenhao) [reply src/support/modify.c:475]: I remember that there is code that we reconstruct the modifies from a tombstone by setting the buffer to empty value. We probably can remove that code as well.

File: src/btree/bt_cursor.c [hunk 3]
```diff
@@ -1745,6 +1746,19 @@ __cursor_chain_exceeded(WT_CURSOR_BTREE *cbt)
...
+     * Don't put a new update on top of an aborted update or non-aborted tombstone. Anything reading
+     * the update chain cannot tell the difference between (1) an aborted set of updates from a
+     * rolled-back transaction, and (2) an in-flight, uncommitted rollback.
+     *
+     * This becomes a problem when trying to construct full values from a sequence of modifies,
+     * since the deltas are calculated using normal visibility rules - but if we can't disambiguate
+     * those two cases, we can't know which transaction the aborted entries belong to, so we can't
+     * calculate a correct delta. (This is only a problem in read-uncommitted isolation.)
+     */
+    if (upd && (upd->txnid == WT_TXN_ABORTED || upd->type == WT_UPDATE_TOMBSTONE))
```
[the remark this answers is not in the document]
Comment (korteland) [reply src/btree/bt_cursor.c]: Thanks - fixed in 04178a5.

File: src/support/modify.c (hunk 1 above)
Comment (korteland) [reply src/support/modify.c]: This turned out to be a rabbit hole - there is other code that can append updates besides what I fixed. So I limited the scope of this ticket to fixing {{__btcur_update}}, which is the function that pushed the bad update in the original test failure. I made WT-12008 for fixing the other code.

File: src/support/modify.c (hunk 2 above)
Comment (korteland) [reply src/support/modify.c]: Also moved this to WT-12008.

**Review by quchenhao — APPROVED** [review]
lgtm

**Review by sulabhM — APPROVED** [review]

**Comment by korteland** [comment]
Perf patch wasn't super interesting - link on ticket.

**Comment by korteland** [comment]
evergreen merge

**Pull request merged**